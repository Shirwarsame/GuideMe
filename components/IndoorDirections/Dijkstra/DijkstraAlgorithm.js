/**
 * Implementation of Dijkstra's shortest path algorithm
 * @param {*} graph | graph of nodes (ClassGraph.js)
 * @param {*} start | starting node
 * @param {*} end | ending node
 */
export const dijkstra = (graph, start, end) => {
    // track the lowest cost to reach each node
    let costs = {};
    costs[end] = "Infinity";
    costs = Object.assign(costs, graph[start]);

    // track paths
    const parents = {end: null};
    for (let child in graph[start]) {
        parents[child] = start;
    }

    const processed = [];
    let node = lowestCostNode(costs, processed);

    while (node) {
        let cost = costs[node];
        let children = graph[node];
        for (let n in children) {
            if (String(n) !== String(start)) {
                let newCost = cost + children[n];
                if (!costs[n] || costs[n] > newCost) {
                    costs[n] = newCost;
                    parents[n] = node;
                }
            }
        }
        processed.push(node);
        node = lowestCostNode(costs, processed);
    }

    let optimalPath = [end];
    let parent = parents[end];
    while (parent) {
        optimalPath.push(parent);
        parent = parents[parent];
    }
    optimalPath.reverse();

    const results = {
        distance: costs[end],
        path: optimalPath
    };

    return results;
};

// Looks for cost of paths & goes with cheapest in cost
const lowestCostNode = (costs, processed) => {
    return Object.keys(costs).reduce((lowest, node) => {
        if (lowest === null || costs[node] < costs[lowest]) {
            if (!processed.includes(node)) {
                lowest = node;
            }
        }
        return lowest;
    }, null);
};


/**
 * Get the floor number of a specific classroom. 
 * Ex: H829 returns 8
 * @param {*} name | classroom name
 */
export const getFloorNumber = (name) => {
    if (name) {
        if (name == "exit" || name == "elevator") {
            return name;
        }

        if (name.match(/\b[^\d\W]+\b/g)) {
            return name;
        }
        
        const num = name.match(/\d+/g).map(Number);
        
        if (name.includes("_")) {
            const room = num[0].toString();
            return room[0];
        }
        if(num.length > 1) {
            return num[0].toString();
        }
        if (num.toString().length == 3) {
            return num.toString().charAt(0)
        }
        if (num.toString().length == 4) {
            return num.toString().charAt(0) + num.toString().charAt(1);
        }   
    }
}

/**
 * Only used to fetch information from data
 * 
 * Change the classroom name to H8XX because our
 * data file stores uses associative arrays with H8XX as the key names. 
 * Ex: H529 returns H829
 * @param {*} name | classroom name
 */
export const ConvertToHall8Floor = (name) => {
    if (!name.includes(" ")) {
        if (name == "exit" || name == "elevator") {
            return name;
        }
        
        if (name) {
            const num = name.match(/\d+/g).map(Number);
            const numToString = num.toString();
            
            if(numToString.charAt(0) == 9) {
                return name;
            }
            else if (numToString.length == 3) {
                return name.replace(numToString.charAt(0), "8");
            }
            else if (numToString.length == 4) {
                return name.replace(numToString.substring(0,2), "8");
            }
        }
    }  
}

/**
 * The following function converts the point of interest
 * to the key value in the data file.
 * @param {*} name 
 */
export const ConvertPointOfInterest = (name) => {
    if (name) {
        switch (name) {
            case "Men's Washroom":
                return "men_washroom"
            case "Women's Washroom":
                return "women_washroom"
        }
    }
}

/**
 * The following function returns the shortest path to
 * the interest point requested by the user.
 * @param {*} graph | node graph
 * @param {*} from | from node
 * @param {*} to | to node
 */
export const shortestPathToInterest = (graph, from, to) => {
    if (to.includes("Washroom")) {
        var to = ConvertPointOfInterest(to);
        return {route: dijkstra(graph, from, to).path, to: to};
    }

    if (to.includes("Water")) {
        var costToNorth = dijkstra(graph, from, "water_foutain_N");
        var costToSouth = dijkstra(graph, from, "water_foutain_S");
        
        if (costToNorth.distance >= costToSouth.distance) {
            return {route: costToSouth.path, to: "water_foutain_S"}
        } else {
            return {route: costToNorth.path, to: "water_foutain_N"}
        }
    }
}
/**
 * The following function returns the coordinates for an arrow
 * to be drawn on the SVG map for a user to follow a direction.
 * @param {*} x1 | x1 coordinate
 * @param {*} y1 | y1 coordinate 
 * @param {*} x2 | x2 coordinate 
 * @param {*} y2 | y2 coordinate 
 */
export const getArrowCoordinates = (x1, y1, x2, y2) => {
    const LINE_DIF = 0.5;
    const THETA = 30;

    var arg1 = (x1 - x2)*Math.cos(THETA*Math.PI/180);
    var arg2 = (y1 - y2)*Math.sin(THETA*Math.PI/180);
    var x3 = (Number(x2) + Number((LINE_DIF*(arg1+arg2))));

    var arg3 = (y1 - y2)*Math.cos(THETA*Math.PI/180);
    var arg4 = (x1 - x2)*Math.sin(THETA*Math.PI/180);
    var y3 = (Number(y2) + Number((LINE_DIF*(arg3-arg4))));

    var x4 = (Number(x2) + Number((LINE_DIF*(arg1-arg2))));
    var y4 = (Number(y2) + Number((LINE_DIF*(arg3+arg4))));

    const results = {
        x3: x3,
        y3: y3,
        x4: x4,
        y4: y4
    };
    return results;
}
