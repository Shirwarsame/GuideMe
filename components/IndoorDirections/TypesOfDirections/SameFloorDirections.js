import React from 'react';
import { Line, G } from 'react-native-svg';

export function SameFloorDirections(props) {
    const rooms = props.rooms;
    const graph = props.graph;
    const route = dijkstra(graph, props.from, props.to).path;

    var getNextRoom = (index) => {
        if (index < route.length){
            return route[++index];
        }
    }

    const lines = [];
    route.forEach((element, index) => {
        if (index < route.length - 1) {
            lines.push({x1: rooms[element].nearestPoint.x, y1: rooms[element].nearestPoint.y, x2: rooms[getNextRoom(index)].nearestPoint.x, y2: rooms[getNextRoom(index)].nearestPoint.y})
        }
    });

    if(props.floor == getFloorNumber(props.to)) {
        return(
            <G>
                <Line x1={rooms[props.from].x} y1={rooms[props.from].y} x2={rooms[props.from].nearestPoint.x} y2={rooms[props.from].nearestPoint.y} stroke="blue" strokeWidth="5"/>
                {
                    lines.map(el => <Line x1={el.x1} y1={el.y1} x2={el.x2} y2={el.y2} stroke="blue" strokeWidth="5"/>)
                }
                <Line x1={rooms[props.to].x} y1={rooms[props.to].y} x2={rooms[props.to].nearestPoint.x} y2={rooms[props.to].nearestPoint.y} stroke="blue" strokeWidth="5"/>
            </G>
        )
    }
    return(<G></G>);
}
const lowestCostNode = (costs, processed) => {
    return Object.keys(costs).reduce((lowest, node) => {
        if (lowest === null || costs[node] < costs[lowest]) {
            if (!processed.includes(node)) {
                lowest = node;
            }
        }
        return lowest;
    }, null);
};

// function that returns the minimum cost and path to reach Finish
const dijkstra = (graph, start, end) => {
    // track the lowest cost to reach each node
    let costs = {};
    costs[end] = "Infinity";
    costs = Object.assign(costs, graph[start]);

    // track paths
    const parents = {end: null};
    for (let child in graph[start]) {
        parents[child] = start;
    }

    const processed = [];
    let node = lowestCostNode(costs, processed);

    while (node) {
        let cost = costs[node];
        let children = graph[node];
        for (let n in children) {
            if (String(n) !== String(start)) {
                let newCost = cost + children[n];
                if (!costs[n] || costs[n] > newCost) {
                    costs[n] = newCost;
                    parents[n] = node;
                }
            }
        }
        processed.push(node);
        node = lowestCostNode(costs, processed);
    }

    let optimalPath = [end];
    let parent = parents[end];
    while (parent) {
        optimalPath.push(parent);
        parent = parents[parent];
    }
    optimalPath.reverse();

    const results = {
        distance: costs[end],
        path: optimalPath
    };

    return results;
};

const getFloorNumber = (name) => {
    const num = name.match(/\d+/g).map(Number);
    return num.toString().charAt(0);
}