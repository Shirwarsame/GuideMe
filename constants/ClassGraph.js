export function ClassGraph () {
    const graph = {
        H801: {H803: 1, H863: 1, H867: 1},
        H803: {H805: 1, H801: 1},
        H805: {H807: 1, H803: 1, women_washroom: 0.5},
        women_washroom: {H805: 0.5},
        H807: {checkpoint1: 1, H805: 1},
        checkpoint1: {H806: 1, H811: 1, H807: 1, exit: 1, water_foutain_S: 1, men_washroom: 1},
        exit: {checkpoint1: 1},
        H806: {checkpoint4: 1, checkpoint1: 1},
        H811: {H813: 1, checkpoint1: 1, men_washroom: 0.5},
        men_washroom: {H811: 0.5, stairs_SW: 1, water_foutain_S: 0.1, checkpoint1: 1}, 
        H813: {H815: 1, H811: 1, stairs_SW: 0.5},
        H815: {H817: 1, H813: 1},
        H817: {H819: 1, H815: 1},
        H819: {H821: 1, H817: 1},
        H821: {H823: 1, H819: 1},
        H823: {H825: 1, H821: 1},
        H825: {H827: 1, H823: 1}, H820: 0.5,
        H827: {H828: 1, H822: 0.5, H825: 1},
        H828: {H829: 1, H827: 1, H822: 1},
        H829: {H831: 0.5, H828: 1},
        H831: {H833: 1, H829: 1},
        H833: {H835: 1, H831: 1, water_foutain_N: 0.5},
        H835: {H837: 1, H833: 1, stairs_NW: 0.5},
        H837: {checkpoint2: 1, H835: 1},
        checkpoint2: {H841: 1, H838: 1, H837: 1, H832: 1, escalator: 3},
        H841: {H843: 1, checkpoint2: 1, H838: 0.5},
        H843: {H845: 1, H841: 1},
        H845: {H847: 1, H843: 1, stairs_NE: 0.5},
        H847: {H849: 1, H845: 1},
        H849: {H851: 1, H847: 1},
        H851: {H853: 1, H849: 1},
        H853: {H851: 1, H855: 1},
        H854: {H855: 0.5},
        H855: {H857: 1, H853: 1, H854: 0.5},
        H857: {H859: 1, H855: 1},
        H859: {checkpoint3: 1, H861: 1, H857: 1},
        H860: {stairs_SE: 0.5, checkpoint3: 1, H862: 1},
        H861: {H863: 1, H859: 1},
        H863: {H867: 1, H801: 1, H861: 1},
        H867: {H863: 1, H801: 1},
        checkpoint3: {H859: 1, H860: 1, stairs_SE: 1},
        H862: {checkpoint4: 2, H860: 1, elevator: 0.5},
        checkpoint4: {H862: 1, H806: 1, elevator: 1, escalator: 1},
        H820: {H825: 0.1, H822: 1},
        H822: {H820: 1, H828: 0.5, H827: 0.5},
        H832: {checkpoint2: 1, H837: 0.5, stairs_NW: 1},
        H838: {checkpoint2: 1, H841: 0.5, escalator: 2},
        elevator: {checkpoint4: 1, stairs_SE: 1, H862: 0.5},
        water_foutain_S: {men_washroom: 0.1, stairs_SW: 1, checkpoint1: 1},
        water_foutain_N: {stairs_NW: 1, H833: 0.5},
        stairs_NE: {H845: 0.5},
        stairs_NW: {H835: 0.5, water_foutain_N: 1, H832: 1},
        stairs_SE: {elevator: 1, H860: 1, checkpoint3: 1},
        stairs_SW: {H813: 0.5, men_washroom: 1, water_foutain_S: 1},
        escalator: {checkpoint2: 3, checkpoint4: 1, H838: 2}
    };
    return graph;
}
