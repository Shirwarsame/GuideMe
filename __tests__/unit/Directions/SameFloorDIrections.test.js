import React from "react";
import renderer from "react-test-renderer";
import { HallXCoordinates } from "../../../constants/HallXCoordinates";
import { SameFloorDirections } from "../../../components/IndoorDirections/TypesOfDirections/SameFloorDirections";
import { getFloorNumber, dijkstra, shortestPathToInterest } from "../../../components/IndoorDirections/Dijkstra/DijkstraAlgorithm";
import { ClassGraph } from "../../../constants/ClassGraph";
import { Class9Graph } from "../../../constants/ClassGraph9";

describe("SameFloorDirections component", () => {
    test("renders correctly", () => {
        const tree = renderer.create(<SameFloorDirections floor={8} rooms={HallXCoordinates()} from={"H835"} to={"H835"}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test("returns correct floor from classroom", () => {
        const testCases = [
            "H829",
            "H531",
            "H123",
            "H1055",
            "H1347",
            "MB3.125",
            "MB12.246",
            "Grey Nuns Annex",
        ]
        const results = [
            "8",
            "5",
            "1",
            "10",
            "13",
            "3",
            "12",
            "Grey Nuns Annex",
        ]
        testCases.forEach((element, index) => {
            expect(getFloorNumber(element)).toEqual(results[index])
        });
    })

    test("get path to closest water fountain", () => {
        const graph = ClassGraph();

        const testCases = [
            {from: "H837", to: "Men's Washroom", graph: graph},
            {from: "H820", to: "Women's Washroom", graph: graph},
            {from: "H815", to: "Water Fountain", graph: graph},
            {from: "H845", to: "Water Fountain", graph: graph},
        ]
        const results = [
            {route: ["H837", "checkpoint2", "escalator", "checkpoint4", "H806", "checkpoint1", "men_washroom"], to: "men_washroom"},
            {route: ["H820", "H825", "H823", "H821", "H819", "H817", "H815", "H813", "H811", "checkpoint1", "H807", "H805", "women_washroom"], to: "women_washroom"},
            {route: ["H815", "H813", "stairs_SW", "water_foutain_S"], to: "water_foutain_S"},
            {route: ["H845", "H843", "H841", "checkpoint2", "H832", "stairs_NW", "water_foutain_N"], to: "water_foutain_N"}
        ]
        testCases.forEach((element, index) => {
            expect(shortestPathToInterest(element.graph, element.from, element.to)).toEqual(results[index]); 
        });
    })

    // The following test ensures the user is guided from a classroom
    // to another classroom in the same floor
    test("dijkstra within the same floor", () => {
        const testCases = [
            {from: "H863", to: "H801"},
            {from: "H803", to: "H827"},
            {from: "H937", to: "H911"},
        ]
        const results = [
            ["H863", "H801"],
            ["H803", "H805", "H807", "checkpoint1", "H811", "H813", "H815", "H817", "H819", "H821", "H823", "H825", "H827"],
            ["H937", "checkpoint3", "checkpoint4", "escalator", "checkpoint7", "H980", "H914","checkpoint1", "men_washroom", "H911"]
        ]

        testCases.forEach((element, index) => {
            const path = dijkstra(getFloorNumber(element.from) == 9 ? Class9Graph() : ClassGraph(), element.from, element.to).path;
            expect(path).toEqual(results[index]);
        });
    })
});
