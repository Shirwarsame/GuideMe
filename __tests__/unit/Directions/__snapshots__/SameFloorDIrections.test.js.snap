// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SameFloorDirections component renders correctly for floor 8 1`] = `
<RNSVGGroup
  fill={4278190080}
  fillOpacity={1}
  fillRule={1}
  font={Object {}}
  matrix={
    Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ]
  }
  opacity={1}
  originX={0}
  originY={0}
  propList={Array []}
  rotation={0}
  scaleX={1}
  scaleY={1}
  skewX={0}
  skewY={0}
  stroke={null}
  strokeDasharray={null}
  strokeDashoffset={null}
  strokeLinecap={0}
  strokeLinejoin={0}
  strokeMiterlimit={4}
  strokeOpacity={1}
  strokeWidth={1}
  vectorEffect={0}
  x={0}
  y={0}
>
  <RNSVGLine
    fill={4278190080}
    fillOpacity={1}
    fillRule={1}
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
    opacity={1}
    originX={0}
    originY={0}
    propList={
      Array [
        "stroke",
        "strokeWidth",
      ]
    }
    rotation={0}
    scaleX={1}
    scaleY={1}
    skewX={0}
    skewY={0}
    stroke={4278190335}
    strokeDasharray={null}
    strokeDashoffset={null}
    strokeLinecap={0}
    strokeLinejoin={0}
    strokeMiterlimit={4}
    strokeOpacity={1}
    strokeWidth="5"
    vectorEffect={0}
    x={0}
    x1="737"
    x2="737"
    y={0}
    y1="844"
    y2="802"
  />
  <RNSVGLine
    fill={4278190080}
    fillOpacity={1}
    fillRule={1}
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
    opacity={1}
    originX={0}
    originY={0}
    propList={
      Array [
        "stroke",
        "strokeWidth",
      ]
    }
    rotation={0}
    scaleX={1}
    scaleY={1}
    skewX={0}
    skewY={0}
    stroke={4278190335}
    strokeDasharray={null}
    strokeDashoffset={null}
    strokeLinecap={0}
    strokeLinejoin={0}
    strokeMiterlimit={4}
    strokeOpacity={1}
    strokeWidth="5"
    vectorEffect={0}
    x={0}
    x1="737"
    x2="737"
    y={0}
    y1="844"
    y2="802"
  />
  <RNSVGLine
    fill={4278190080}
    fillOpacity={1}
    fillRule={1}
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
    opacity={1}
    originX={0}
    originY={0}
    propList={
      Array [
        "stroke",
        "strokeWidth",
      ]
    }
    rotation={0}
    scaleX={1}
    scaleY={1}
    skewX={0}
    skewY={0}
    stroke={4278190335}
    strokeDasharray={null}
    strokeDashoffset={null}
    strokeLinecap={0}
    strokeLinejoin={0}
    strokeMiterlimit={4}
    strokeOpacity={1}
    strokeWidth="5"
    vectorEffect={0}
    x={0}
    x1={726.5}
    x2="737"
    y={0}
    y1={825.8134665205268}
    y2="844"
  />
  <RNSVGLine
    fill={4278190080}
    fillOpacity={1}
    fillRule={1}
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
    opacity={1}
    originX={0}
    originY={0}
    propList={
      Array [
        "stroke",
        "strokeWidth",
      ]
    }
    rotation={0}
    scaleX={1}
    scaleY={1}
    skewX={0}
    skewY={0}
    stroke={4278190335}
    strokeDasharray={null}
    strokeDashoffset={null}
    strokeLinecap={0}
    strokeLinejoin={0}
    strokeMiterlimit={4}
    strokeOpacity={1}
    strokeWidth="5"
    vectorEffect={0}
    x={0}
    x1={747.5}
    x2="737"
    y={0}
    y1={825.8134665205268}
    y2="844"
  />
</RNSVGGroup>
`;

exports[`SameFloorDirections component renders correctly for floor 9 1`] = `
<RNSVGGroup
  fill={4278190080}
  fillOpacity={1}
  fillRule={1}
  font={Object {}}
  matrix={
    Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ]
  }
  opacity={1}
  originX={0}
  originY={0}
  propList={Array []}
  rotation={0}
  scaleX={1}
  scaleY={1}
  skewX={0}
  skewY={0}
  stroke={null}
  strokeDasharray={null}
  strokeDashoffset={null}
  strokeLinecap={0}
  strokeLinejoin={0}
  strokeMiterlimit={4}
  strokeOpacity={1}
  strokeWidth={1}
  vectorEffect={0}
  x={0}
  y={0}
>
  <RNSVGLine
    fill={4278190080}
    fillOpacity={1}
    fillRule={1}
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
    opacity={1}
    originX={0}
    originY={0}
    propList={
      Array [
        "stroke",
        "strokeWidth",
      ]
    }
    rotation={0}
    scaleX={1}
    scaleY={1}
    skewX={0}
    skewY={0}
    stroke={4278190335}
    strokeDasharray={null}
    strokeDashoffset={null}
    strokeLinecap={0}
    strokeLinejoin={0}
    strokeMiterlimit={4}
    strokeOpacity={1}
    strokeWidth="5"
    vectorEffect={0}
    x={0}
    x1="751"
    x2="832"
    y={0}
    y1="362"
    y2="381"
  />
  <RNSVGLine
    fill={4278190080}
    fillOpacity={1}
    fillRule={1}
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
    opacity={1}
    originX={0}
    originY={0}
    propList={
      Array [
        "stroke",
        "strokeWidth",
      ]
    }
    rotation={0}
    scaleX={1}
    scaleY={1}
    skewX={0}
    skewY={0}
    stroke={4278190335}
    strokeDasharray={null}
    strokeDashoffset={null}
    strokeLinecap={0}
    strokeLinejoin={0}
    strokeMiterlimit={4}
    strokeOpacity={1}
    strokeWidth="5"
    vectorEffect={0}
    x={0}
    x1="832"
    x2="832"
    y={0}
    y1="381"
    y2="381"
  />
  <RNSVGLine
    fill={4278190080}
    fillOpacity={1}
    fillRule={1}
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
    opacity={1}
    originX={0}
    originY={0}
    propList={
      Array [
        "stroke",
        "strokeWidth",
      ]
    }
    rotation={0}
    scaleX={1}
    scaleY={1}
    skewX={0}
    skewY={0}
    stroke={4278190335}
    strokeDasharray={null}
    strokeDashoffset={null}
    strokeLinecap={0}
    strokeLinejoin={0}
    strokeMiterlimit={4}
    strokeOpacity={1}
    strokeWidth="5"
    vectorEffect={0}
    x={0}
    x1="832"
    x2="832"
    y={0}
    y1="381"
    y2="285"
  />
  <RNSVGLine
    fill={4278190080}
    fillOpacity={1}
    fillRule={1}
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
    opacity={1}
    originX={0}
    originY={0}
    propList={
      Array [
        "stroke",
        "strokeWidth",
      ]
    }
    rotation={0}
    scaleX={1}
    scaleY={1}
    skewX={0}
    skewY={0}
    stroke={4278190335}
    strokeDasharray={null}
    strokeDashoffset={null}
    strokeLinecap={0}
    strokeLinejoin={0}
    strokeMiterlimit={4}
    strokeOpacity={1}
    strokeWidth="5"
    vectorEffect={0}
    x={0}
    x1="832"
    x2="822"
    y={0}
    y1="285"
    y2="170"
  />
  <RNSVGLine
    fill={4278190080}
    fillOpacity={1}
    fillRule={1}
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
    opacity={1}
    originX={0}
    originY={0}
    propList={
      Array [
        "stroke",
        "strokeWidth",
      ]
    }
    rotation={0}
    scaleX={1}
    scaleY={1}
    skewX={0}
    skewY={0}
    stroke={4278190335}
    strokeDasharray={null}
    strokeDashoffset={null}
    strokeLinecap={0}
    strokeLinejoin={0}
    strokeMiterlimit={4}
    strokeOpacity={1}
    strokeWidth="5"
    vectorEffect={0}
    x={0}
    x1="822"
    x2="822"
    y={0}
    y1="170"
    y2="170"
  />
  <RNSVGLine
    fill={4278190080}
    fillOpacity={1}
    fillRule={1}
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
    opacity={1}
    originX={0}
    originY={0}
    propList={
      Array [
        "stroke",
        "strokeWidth",
      ]
    }
    rotation={0}
    scaleX={1}
    scaleY={1}
    skewX={0}
    skewY={0}
    stroke={4278190335}
    strokeDasharray={null}
    strokeDashoffset={null}
    strokeLinecap={0}
    strokeLinejoin={0}
    strokeMiterlimit={4}
    strokeOpacity={1}
    strokeWidth="5"
    vectorEffect={0}
    x={0}
    x1="822"
    x2="822"
    y={0}
    y1="170"
    y2="170"
  />
  <RNSVGLine
    fill={4278190080}
    fillOpacity={1}
    fillRule={1}
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
    opacity={1}
    originX={0}
    originY={0}
    propList={
      Array [
        "stroke",
        "strokeWidth",
      ]
    }
    rotation={0}
    scaleX={1}
    scaleY={1}
    skewX={0}
    skewY={0}
    stroke={4278190335}
    strokeDasharray={null}
    strokeDashoffset={null}
    strokeLinecap={0}
    strokeLinejoin={0}
    strokeMiterlimit={4}
    strokeOpacity={1}
    strokeWidth="5"
    vectorEffect={0}
    x={0}
    x1="822"
    x2="714"
    y={0}
    y1="170"
    y2="170"
  />
  <RNSVGLine
    fill={4278190080}
    fillOpacity={1}
    fillRule={1}
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
    opacity={1}
    originX={0}
    originY={0}
    propList={
      Array [
        "stroke",
        "strokeWidth",
      ]
    }
    rotation={0}
    scaleX={1}
    scaleY={1}
    skewX={0}
    skewY={0}
    stroke={4278190335}
    strokeDasharray={null}
    strokeDashoffset={null}
    strokeLinecap={0}
    strokeLinejoin={0}
    strokeMiterlimit={4}
    strokeOpacity={1}
    strokeWidth="5"
    vectorEffect={0}
    x={0}
    x1="714"
    x2="634"
    y={0}
    y1="170"
    y2="170"
  />
  <RNSVGLine
    fill={4278190080}
    fillOpacity={1}
    fillRule={1}
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
    opacity={1}
    originX={0}
    originY={0}
    propList={
      Array [
        "stroke",
        "strokeWidth",
      ]
    }
    rotation={0}
    scaleX={1}
    scaleY={1}
    skewX={0}
    skewY={0}
    stroke={4278190335}
    strokeDasharray={null}
    strokeDashoffset={null}
    strokeLinecap={0}
    strokeLinejoin={0}
    strokeMiterlimit={4}
    strokeOpacity={1}
    strokeWidth="5"
    vectorEffect={0}
    x={0}
    x1="634"
    x2="634"
    y={0}
    y1="125"
    y2="170"
  />
  <RNSVGLine
    fill={4278190080}
    fillOpacity={1}
    fillRule={1}
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
    opacity={1}
    originX={0}
    originY={0}
    propList={
      Array [
        "stroke",
        "strokeWidth",
      ]
    }
    rotation={0}
    scaleX={1}
    scaleY={1}
    skewX={0}
    skewY={0}
    stroke={4278190335}
    strokeDasharray={null}
    strokeDashoffset={null}
    strokeLinecap={0}
    strokeLinejoin={0}
    strokeMiterlimit={4}
    strokeOpacity={1}
    strokeWidth="5"
    vectorEffect={0}
    x={0}
    x1={645.25}
    x2="634"
    y={0}
    y1={144.48557158514987}
    y2="125"
  />
  <RNSVGLine
    fill={4278190080}
    fillOpacity={1}
    fillRule={1}
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
    opacity={1}
    originX={0}
    originY={0}
    propList={
      Array [
        "stroke",
        "strokeWidth",
      ]
    }
    rotation={0}
    scaleX={1}
    scaleY={1}
    skewX={0}
    skewY={0}
    stroke={4278190335}
    strokeDasharray={null}
    strokeDashoffset={null}
    strokeLinecap={0}
    strokeLinejoin={0}
    strokeMiterlimit={4}
    strokeOpacity={1}
    strokeWidth="5"
    vectorEffect={0}
    x={0}
    x1={622.75}
    x2="634"
    y={0}
    y1={144.48557158514987}
    y2="125"
  />
</RNSVGGroup>
`;

exports[`SameFloorDirections component renders correctly for interest 1`] = `
<RNSVGGroup
  fill={4278190080}
  fillOpacity={1}
  fillRule={1}
  font={Object {}}
  matrix={
    Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ]
  }
  opacity={1}
  originX={0}
  originY={0}
  propList={Array []}
  rotation={0}
  scaleX={1}
  scaleY={1}
  skewX={0}
  skewY={0}
  stroke={null}
  strokeDasharray={null}
  strokeDashoffset={null}
  strokeLinecap={0}
  strokeLinejoin={0}
  strokeMiterlimit={4}
  strokeOpacity={1}
  strokeWidth={1}
  vectorEffect={0}
  x={0}
  y={0}
/>
`;

exports[`SameFloorDirections component renders correctly for loyola 1`] = `
<RNSVGGroup
  fill={4278190080}
  fillOpacity={1}
  fillRule={1}
  font={Object {}}
  matrix={
    Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ]
  }
  opacity={1}
  originX={0}
  originY={0}
  propList={Array []}
  rotation={0}
  scaleX={1}
  scaleY={1}
  skewX={0}
  skewY={0}
  stroke={null}
  strokeDasharray={null}
  strokeDashoffset={null}
  strokeLinecap={0}
  strokeLinejoin={0}
  strokeMiterlimit={4}
  strokeOpacity={1}
  strokeWidth={1}
  vectorEffect={0}
  x={0}
  y={0}
>
  <RNSVGLine
    fill={4278190080}
    fillOpacity={1}
    fillRule={1}
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
    opacity={1}
    originX={0}
    originY={0}
    propList={
      Array [
        "stroke",
        "strokeWidth",
      ]
    }
    rotation={0}
    scaleX={1}
    scaleY={1}
    skewX={0}
    skewY={0}
    stroke={4278190335}
    strokeDasharray={null}
    strokeDashoffset={null}
    strokeLinecap={0}
    strokeLinejoin={0}
    strokeMiterlimit={4}
    strokeOpacity={1}
    strokeWidth="5"
    vectorEffect={0}
    x={0}
    x1={398}
    x2={398}
    y={0}
    y1={439}
    y2={497}
  />
  <RNSVGLine
    fill={4278190080}
    fillOpacity={1}
    fillRule={1}
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
    opacity={1}
    originX={0}
    originY={0}
    propList={
      Array [
        "stroke",
        "strokeWidth",
      ]
    }
    rotation={0}
    scaleX={1}
    scaleY={1}
    skewX={0}
    skewY={0}
    stroke={4278190335}
    strokeDasharray={null}
    strokeDashoffset={null}
    strokeLinecap={0}
    strokeLinejoin={0}
    strokeMiterlimit={4}
    strokeOpacity={1}
    strokeWidth="5"
    vectorEffect={0}
    x={0}
    x1={398}
    x2={135}
    y={0}
    y1={497}
    y2={892}
  />
  <RNSVGLine
    fill={4278190080}
    fillOpacity={1}
    fillRule={1}
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
    opacity={1}
    originX={0}
    originY={0}
    propList={
      Array [
        "stroke",
        "strokeWidth",
      ]
    }
    rotation={0}
    scaleX={1}
    scaleY={1}
    skewX={0}
    skewY={0}
    stroke={4278190335}
    strokeDasharray={null}
    strokeDashoffset={null}
    strokeLinecap={0}
    strokeLinejoin={0}
    strokeMiterlimit={4}
    strokeOpacity={1}
    strokeWidth="5"
    vectorEffect={0}
    x={0}
    x1={61}
    x2={135}
    y={0}
    y1={892}
    y2={892}
  />
  <RNSVGLine
    fill={4278190080}
    fillOpacity={1}
    fillRule={1}
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
    opacity={1}
    originX={0}
    originY={0}
    propList={
      Array [
        "stroke",
        "strokeWidth",
      ]
    }
    rotation={0}
    scaleX={1}
    scaleY={1}
    skewX={0}
    skewY={0}
    stroke={4278190335}
    strokeDasharray={null}
    strokeDashoffset={null}
    strokeLinecap={0}
    strokeLinejoin={0}
    strokeMiterlimit={4}
    strokeOpacity={1}
    strokeWidth="5"
    vectorEffect={0}
    x={0}
    x1={93.04293994002424}
    x2={61}
    y={0}
    y1={873.5}
    y2={892}
  />
  <RNSVGLine
    fill={4278190080}
    fillOpacity={1}
    fillRule={1}
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
    opacity={1}
    originX={0}
    originY={0}
    propList={
      Array [
        "stroke",
        "strokeWidth",
      ]
    }
    rotation={0}
    scaleX={1}
    scaleY={1}
    skewX={0}
    skewY={0}
    stroke={4278190335}
    strokeDasharray={null}
    strokeDashoffset={null}
    strokeLinecap={0}
    strokeLinejoin={0}
    strokeMiterlimit={4}
    strokeOpacity={1}
    strokeWidth="5"
    vectorEffect={0}
    x={0}
    x1={93.04293994002424}
    x2={61}
    y={0}
    y1={910.5}
    y2={892}
  />
</RNSVGGroup>
`;
